### PR TITLE
python312Packages.busylight-for-humans: init at 0.32.0

### PR DESCRIPTION
**Pull request by Conni2461**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Scriptkiddi** [comment]
@NixOS/nixpkgs-merge-bot merge

File: pkgs/development/python-modules/bitvector-for-humans/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  poetry-core,
+  pytestCheckHook,
+}:
+buildPythonPackage rec {
+  pname = "bitvector-for-humans";
+  version = "0.14.1";
+  format = "pyproject";
```
Comment (mweinelt) [inline pkgs/development/python-modules/bitvector-for-humans/default.nix:11]:
```suggestion
  pyproject = true
```

File: pkgs/development/python-modules/bitvector-for-humans/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail 'poetry>=0.12' 'poetry-core' \
+      --replace-fail 'poetry.masonry.api' 'poetry.core.masonry.api'
+  '';
+
+  nativeBuildInputs = [
+    poetry-core
+    pytestCheckHook
+  ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/bitvector-for-humans/default.nix:29]:
```suggestion
  build-system = [ poetry-core ];
  
  nativeCheckInputs = [ pytestCheckHook ];
```

File: pkgs/development/python-modules/bitvector-for-humans/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    owner = "JnyJny";
+    repo = "bitvector";
+    rev = version;
+    hash = "sha256-GVTRD83tq/Hea53US4drOD5ruoYCLTVd24aZOSdDsSo=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail 'poetry>=0.12' 'poetry-core' \
+      --replace-fail 'poetry.masonry.api' 'poetry.core.masonry.api'
```
Comment (mweinelt) [inline pkgs/development/python-modules/bitvector-for-humans/default.nix:23]: Please send this change upstream.

File: pkgs/development/python-modules/busylight-for-humans/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail 'poetry>=0.12' 'poetry-core' \
+      --replace-fail 'poetry.masonry.api' 'poetry.core.masonry.api'
+  '';
+
+  nativeBuildInputs = [
+    poetry-core
+    pytestCheckHook
+    pytest-mock
```
Comment (mweinelt) [inline pkgs/development/python-modules/busylight-for-humans/default.nix:36]: These two are test dependencies, they go into `nativeCheckInputs`.

File: pkgs/development/python-modules/busylight-for-humans/default.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  ];
+  propagatedBuildInputs = [
```
Comment (mweinelt) [inline pkgs/development/python-modules/busylight-for-humans/default.nix:38]:
```suggestion
  dependencies = [
```

Newer lingo for propagatedBuildInputs in python packages.

File: pkgs/development/python-modules/busylight-for-humans/default.nix [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    webcolors
+  ];
+
+  disabledTestPaths = [ "tests/test_pydantic_models.py" ];
+
+  pythonImportsCheck = [ "busylight" ];
+
+  meta = with lib; {
+    homepage = "https://github.com/JnyJny/busylight";
+    description = "Control USB connected presence lights from multiple vendors via the command-line or web API.";
```
Comment (mweinelt) [inline pkgs/development/python-modules/busylight-for-humans/default.nix:53]: missing mainProgram

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by 0x4A6F** [comment]
Could you also utilize `busylight udev-rules` to generate these [udev Rules](https://github.com/JnyJny/busylight?tab=readme-ov-file#linux-post-install-activities) and provide it from this package?

**Review by 0x4A6F — APPROVED** [review]

**Review by mweinelt — APPROVED** [review]

**Pull request merged**